cmd/admin-handlers.go

```diff
@@ -1303,6 +1304,155 @@ func (a adminAPIHandlers) KMSKeyStatusHandler(w http.ResponseWriter, r *http.Req
 	writeSuccessResponseJSON(w, resp)
 }
 
+// KMSKeyRotateHandler - POST /minio/admin/v2/kms/key/rotate
+func (a adminAPIHandlers) KMSKeyRotateHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "KMSKeyRotateHandler")
+
+	objectAPI, _ := validateAdminReq(ctx, w, r, iampolicy.KMSKeyRotateAdminAction)
+	if objectAPI == nil {
+		return
+	}
+
+	if GlobalKMS == nil {
+		writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrKMSNotConfigured), r.URL)
+		return
+	}
+
+	var req madmin.KMSRotateKeyRequest
+	if err := json.NewDecoder(io.LimitReader(r.Body, 1<<20)).Decode(&req); err != nil {
+		writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
+		return
+	}
+
+	// Delete rotation entries where the old key ID
+	// is equal to the new key ID. It's pointless to
+	// rotate when the old and the new key are the same.
+	for k, v := range req.KeyMapping {
+		if k == v {
+			delete(req.KeyMapping, k)
+		}
+	}
+
+	// Verify that the key mapping is well-formed. In particular,
+	// that a old key ID does not appear as new key ID as well.
+	// This is not allowed since it would cause unpredictable
+	// behavior.
+	for oldKey := range req.KeyMapping {
+		for _, newKey := range req.KeyMapping {
+			if oldKey == newKey {
+				err := AdminError{
+					StatusCode: http.StatusBadRequest,
+					Code:       http.StatusText(http.StatusBadRequest),
```

Comment:
Here the code can be `XMinIOEncryptionError`  perhaps it can be `err := crypto.Errorf("invalid key mapping: cannot rotate from and to: %s", oldKey)`